be/src/exprs/array_functions.tpp

```diff
@@ -101,26 +101,25 @@ private:
         Datum v = column.get(index);
         const auto& items = v.get<DatumArray>();
 
+        auto& dest_data_column = dest_column->elements_column();
+        auto& dest_offsets = dest_column->offsets_column()->get_data();
+
         for (const auto& item : items) {
             if (item.is_null()) {
-                has_null = true;
-            } else {
-                hash_set->emplace(item.get<CppType>());
+                if (!has_null) {
+                    dest_data_column->append_nulls(1);
+                    has_null = true;
+                }
+                continue;
             }
-        }
-
-        auto& dest_data_column = dest_column->elements_column();
-        auto& dest_offsets = dest_column->offsets_column()->get_data();
 
-        if (has_null) {
-            dest_data_column->append_nulls(1);
+            const auto& tt = item.get<CppType>();
+            if (hash_set->count(tt) == 0) {
```

Comment:
does count() should iterator all of element in the set? seems not efficient

Reply:
i think it's not key-point in th function, and should be better than before